### PR TITLE
Refactored server code to prepare for dvv module

### PR DESCRIPTION
**Pull request by andyttran**

### CODE REVIEW

File: server/dvv.js [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+  //Start the server
+  server.listen(port, function() {
+    console.log('Server listening on port ' + port);
+  });
+  
+  //Tell express where to serve static files from
+  app.use(express.static(__dirname + STATIC_PATH));
+
+  //TODO: tell express where to find the favicon
+  //app.use(favicon(__dirname + STATIC_PATH));
```
Comment (ekong2) [inline server/dvv.js]: someone want to do the favicon?

File: server/dvv.js [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+      //If the computation was completed successfully 
+      //and it was delivered back in a timely fashion,
+      //add it to the completed packets heap/ remove from pending packets
+      //**NOTE: this implies late packages will be chucked aside
+      if (data.id !== -1 && pendingPackets[data.id]){
+        delete pendingPackets[data.id];
+        completedPackets.insert(data);
+        
+        //Update everyone on the current progress
+        //TODO: perhaps modify to do it only every once in a while
```
Comment (ekong2) [inline server/dvv.js]: maybe a granular progress might be good? Doesn't seem necessary unless there are a ton of packets

Comment (andyttran) [inline server/dvv.js]: agreed possibly have a minimum increment value

File: server/dvv.js [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+  function initializeProcess(partitionedData){
+    var counter = 0;
+    partitionedData.forEach(function(element){
+      var packet = {'id': counter++, 'payload': element};
+      unsentPackets.insert(packet);
+    });
+  }
+
+  //Reset all data structures
+  function resetProcess(){
```
Comment (ekong2) [inline server/dvv.js]: needs more guidance

File: server/dvv.js [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+    availableClients = [];
+    unsentPackets = new MinHeap();
+    pendingPackets = {};
+    completedPackets = new MinHeap();
+  }
+
+  //Send the next available packet
+  function sendNextAvailablePacket(socket){
```
Comment (ekong2) [inline server/dvv.js]: Definitely need to comment more here as well

File: server/server.js [hunk 5]
```diff
@@ -54,152 +21,37 @@ var createMatrixArrays = function(matrixSize, arrayLength){
...
-    socket.broadcast.emit('clientChange', { 
-      availableClients : availableClients.length 
-    });
-	});
+//Test 1 : default test case
+dvv.config({
+  staticPath: '/../client',
+  timeout: 10000,
+  clock: true
 });
```
Comment (ekong2) [inline server/server.js]: add a first test case? a doubler?

File: server/dvv.js [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+    PARTITION = params.partition;
+  }
+
+  if('partitionLength' in params){
+    PARTITION_LENGTH = params.partitionLength;
+  }
+
+  if('callback' in params){
+    CALLBACK = params.callback;
+  }
```
Comment (ekong2) [inline server/dvv.js]: spacing

File: server/dvv.js (hunk 1 above)
Comment (andyttran) [inline server/dvv.js]: Nathan likely will be the one to tackle this since he is taking care of the logo

File: server/dvv.js (hunk 3 above)
Comment (andyttran) [inline server/dvv.js]: Completed

File: server/dvv.js (hunk 4 above)
Comment (andyttran) [inline server/dvv.js]: Description of data structures will be made on the instantiation only, not reset

File: server/server.js (hunk 5 above)
Comment (andyttran) [inline server/server.js]: The first test case tests only the default values preset in dvv module.
We can add another case in the documentation.

File: server/dvv.js (hunk 6 above)
Comment (andyttran) [inline server/dvv.js]: would you rather have the spaces removed?

**Pull request merged**